include/rapidjson/document.h

```diff
@@ -1749,7 +1749,22 @@ class GenericDocument : public GenericValue<Encoding, Allocator> {
     typedef Allocator AllocatorType;                        //!< Allocator type from template parameter.
 
     //! Constructor
-    /*! \param allocator        Optional allocator for allocating memory.
+    /*! Creates an empty document of specified type.
+        \param type             Mandatory type of object to create.
+        \param allocator        Optional allocator for allocating memory.
+        \param stackCapacity    Optional initial capacity of stack in bytes.
+        \param stackAllocator   Optional allocator for allocating memory for stack.
+    */
+    GenericDocument(Type type, Allocator* allocator = 0, size_t stackCapacity = kDefaultStackCapacity, StackAllocator* stackAllocator = 0) :

```

Comment:
I would recommend to mark this constructor as `explicit` to avoid accidentally creating a temporary `GenericDocument` from a `Type` enum value (because all arguments but the first one are optional).

Strictly speaking, the same applies to the "default" constructor below, but is less important as allocator pointers may not be used as frequently in user code.